app/controllers/vendors_controller.rb

```diff
@@ -23,9 +26,25 @@ def create
     end
   end
 
+  def update
+    @vendor = Vendor.find(params[:id])
+    if Vendor.approve(@vendor)
+      flash[:notice] = "Vendor Updated"
+      redirect_to vendors_path
+    else
+      flash[:notice] = "Error"
+      render vendors_path
+    end
+  end
```

Comment:
Also, it's worth noting that the way it looks to me right now, this update action *only* allows the approval functionality. Right now, I don't see any functionality that allows someone to actually edit the details of a vendor. That doesn't need to be taken care of in this PR, but do note that you should incorporate that in the future.